samples/multiplatform/kmp-ios-app/kmp-ios-app/RepositoryManager.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  RepositoryManager.swift
+//  kmp-ios-app
+//
+//  Created by Ellen Shapiro on 4/26/20.
+//
+
+import Foundation
+import kmp_lib_sample
+import SwiftUI
+
+class RepositoryManager: ObservableObject {
+    
+    private let repository = ApolloiOSRepositoryKt.create()
+    
+    @Published var repos: [RepositoryFragment] = []
+    @Published var repoDetails: [String: RepositoryDetail] = [:]
+    @Published var commits: [String: [GithubRepositoryCommitsQuery.Node]] = [:]
+    
+    func fetch() {
+        self.repository.fetchRepositories { [weak self] repos in
+            self?.repos = repos
+        }
+    }
+    
+    func fetchDetails(for repo: RepositoryFragment) {        
+        self.repository.fetchRepositoryDetail(repositoryName: repo.name) { [weak self] detail in
+            if let detail = detail {
+                self?.repoDetails[repo.name] = detail
+            }
+        }
+    }
+    
+    func fetchCommits(for repo: RepositoryFragment) {
+        // NOTE: This comes in as `[Any]` due to some some issues with representing
+        // optional types in a generic array in Objective-C from K/N. The actual
+        // type coming back here is `GithubRepositoryCommitsQuery.Edge`, and the
+        // `Node` it contains is where actual information about the commit lives.
+        self.repository.fetchCommits(repositoryName: repo.name) { [weak self] commits in
+            let filteredCommits = commits
+                .compactMap { $0 as? GithubRepositoryCommitsQuery.Edge }
```

Comment:
Is there a way to force a crash if `$0` is not nil but not of type `GithubRepositoryCommitsQuery.Edge` either? Some swift equivalent of `compactMap { $0 as GithubRepositoryCommitsQuery.Edge? }` in Kotlin?

Reply:
🤔 Good question - this isn't really a pattern used much in iOS so I don't think it's a change worth making generally, but I'll mess around and see because now I'm curious :)

Reply:
OK, survey says that's not a thing in Swift: 
<img width="934" alt="Screen Shot 2020-04-30 at 5 23 42 PM" src="https://user-images.githubusercontent.com/1976498/80764945-6c866380-8b07-11ea-9f49-dd98c83f0f9f.png">

You also can't do`$0 as? GithubRepositoryCommitsQuery.Edge?` since the type is `[Any]` rather than `[Any?]`: 

<img width="1005" alt="Screen Shot 2020-04-30 at 5 30 19 PM" src="https://user-images.githubusercontent.com/1976498/80765314-588f3180-8b08-11ea-913d-f650e60f733d.png">

If the type were `[Any?]` then you could do that: 

<img width="804" alt="Screen Shot 2020-04-30 at 5 29 03 PM" src="https://user-images.githubusercontent.com/1976498/80765373-7ceb0e00-8b08-11ea-8c7c-80bfa77d646b.png">

But yeah basically you can't do what causes a runtime crash in Kotlin because it causes a compile-time error in Swift.